File: RealtyReachApi/Models/Professional.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RealtyReachApi.Models;

public class Professional
{
    public Guid Id { get; set; }
    public int ProfessionalTypeId { get; set; } // Foreign Key
    
    // Navigation properties
    //public ProfessionalType? ProfessionalType { get; set; }
    public ICollection<JobProfessionalLink>? JobProfessionalLinks { get; set; } = new List<JobProfessionalLink>();
    
    public string? Email { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? ABN { get; set; } // Australian Business Number
    public string? LicenseNumber { get; set; }
    public bool VerificationStatus { get; set; } // True if verified, false otherwise
    public string? CompanyName { get; set; }
    //public List<int>  SuburbIds { get; set; } = new List<int>();
    public List<string>? Regions { get; set; }
    public List<AustralianState>? States { get; set; }
    public List<Specialisation>? Specialisations { get; set; }
    public bool? FirstLogin { get; set; }
}

File: frontend/realty-reach-main/src/Models/Professional.ts
import { AustralianState, Specialisation } from './Job';

/**
 * Base professional data interface
 * Used for basic professional information from /api/Professional endpoint
 */
export interface Professional {
    id: string;
    professionalTypeId: number;
    email?: string;
    firstName?: string;
    lastName?: string;
    abn?: string;
    licenseNumber?: string;
    verificationStatus: boolean;
    companyName?: string;
    regions?: string[];
    states?: AustralianState[];
    specialisations?: Specialisation[];
    firstLogin?: boolean;
    professionalType?: string;
}

/**
 * Extended professional profile data with professional type details
 * Returned from the /api/Professional/profile endpoint
 */
export interface ProfessionalProfileData extends Professional {
    professionalTypeName: string;
    professionalTypeDescription: string;
    regions: string[];
    states: number[];
    specialisations: number[];
}

/**
 * Professional profile with additional metrics
 * Used for display purposes with calculated fields
 */
export interface ProfessionalProfileWithMetrics extends Professional {
    // Additional fields for the profile view
    rating?: number;
    reviewCount?: number;
    completedJobs?: number;
    responseRate?: number;
    averageResponseTime?: number;
}